File: backend/models/api/season.go
package api

import dbModels "github.com/STREAM-BUSTER/stream-buster/models/db"

type Season struct {
	MediaID uint
	Media   *dbModels.Media

	SeasonTMDBID string
	SeasonNumber int
	EpisodeCount int

	Name     string
	Overview string

	PosterPath string
	Episodes   []*Episode
}

File: frontend/src/models/season.ts
import { Media } from './media';
import { Episode } from './episode';  // Assuming an Episode interface

export interface Season {
  mediaID: number;
  media?: Media;  // Optional field
  name: string;
  overview: string;
  seasonTMDBID: string;
  seasonNumber: number;
  posterPath: string;
  episodes: Episode[];  // Array of Episodes
}